climada/engine/uncertainty/unc_cost_benefit.py

```diff
@@ -0,0 +1,227 @@
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define Uncertainty Cost Benefit class
+"""
+
+__all__ = ['UncCostBenefit']
+
+import logging
+import time
+from functools import partial
+import pandas as pd
+
+from climada.engine.uncertainty.base import Uncertainty, UncVar
+from climada.engine.cost_benefit import CostBenefit
+from climada.util import log_level
+
+LOGGER = logging.getLogger(__name__)
+
+# Future planed features:
+# - Add 'efc' (frequency curve) to UncCostBenenfit
+
+class UncCostBenefit(Uncertainty):
+    """
+    Cost Benefit Uncertainty analysis class
+
+    This is the base class to perform uncertainty analysis on the outputs of a
+    climada.engine.costbenefit.CostBenefit() object using the SALib package.
+
+    """
+
+    def __init__(self, haz_unc, ent_unc, haz_fut_unc=None, ent_fut_unc=None):
+        """Initialize UncCostBenefit
+
+        Parameters
+        ----------
+        haz_unc : climada.engine.uncertainty.UncVar
+                  or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard for the present Hazard
+            in climada.engine.CostBenefit.calc
+        ent_unc : climada.engine.uncertainty.UncVar
+                  or climada.entity.Entity
+            Entity uncertainty variable or Entity for the future Entity
+            in climada.engine.CostBenefit.calc
+        haz_unc_fut: climada.engine.uncertainty.UncVar
+                     or climada.hazard.Hazard, optional
+            Hazard uncertainty variable or Hazard for the future Hazard
+            in climada.engine.CostBenefit.calc
+            The Default is None.
+        ent_fut_unc : climada.engine.uncertainty.UncVar
+                      or climada.entity.Entity, optional
+            Entity uncertainty variable or Entity for the future Entity
+            in climada.engine.CostBenefit.calc
+
+        """
+
+        unc_vars = {
+            'haz': UncVar.var_to_uncvar(haz_unc),
+            'ent': UncVar.var_to_uncvar(ent_unc),
+            'haz_fut': UncVar.var_to_uncvar(haz_fut_unc),
+            'ent_fut': UncVar.var_to_uncvar(ent_fut_unc)
+            }
+
+        metrics =  {
+            'tot_climate_risk': pd.DataFrame([]),
+            'benefit': pd.DataFrame([]),
+            'cost_ben_ratio': pd.DataFrame([]),
+            'imp_meas_present': pd.DataFrame([]),
+            'imp_meas_future': pd.DataFrame([])
+            }
```

Comment:
no cost uncertainty by default?

Reply:
This is in the `imp_meas_present` and `imp_meas_future`, c.f. https://github.com/CLIMADA-project/climada_python/blob/26da4ccb775f38e52df2e9022ebf6e96e61ded64/climada/engine/cost_benefit.py#L104